src/mlpack/methods/cf/normalization/item_mean_normalization.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file item_mean_normalization.hpp
+ * @author Wenhao Huang
+ *
+ * This class performs item mean normalization on raw ratings. In another
+ * word, this class is used to remove global effect of item mean.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_NORMALIZATION_ITEM_MEAN_NORMALIZATION_HPP
+#define MLPACK_METHODS_CF_NORMALIZATION_ITEM_MEAN_NORMALIZATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * This normalization class performs item mean normalization on raw ratings.
+ *
+ * An example of how to use ItemMeanNormalization in CF is shown below:
+ *
+ * @code
+ * extern arma::mat data; // (user, item, rating) table
+ * extern arma::Col<size_t> users; // users seeking recommendations
+ * arma::Mat<size_t> recommendations; // Recommendations
+ *
+ * CFType<ItemMeanNormalization> cf(data);
+ *
+ * // Generate 10 recommendations for all users.
+ * cf.GetRecommendations(10, recommendations);
+ * @endcode
+ */
+class ItemMeanNormalization
+{
+ public:
+  // Empty constructor.
+  ItemMeanNormalization() { }
+
+  /**
+   * Normalize the data by subtracting item mean from each of existing ratings.
+   *
+   * @param data Input dataset in the form of coordinate list.
+   */
+  void Normalize(arma::mat& data)
+  {
+    const size_t itemNum = arma::max(data.row(1)) + 1;
+    itemMean = arma::vec(itemNum, arma::fill::zeros);
+    // Number of ratings for each item.
+    arma::Row<size_t> ratingNum(itemNum, arma::fill::zeros);
+
+    // Sum ratings for each item.
+    data.each_col([&](arma::vec& datapoint) {
+      const size_t item = (size_t) datapoint(1);
+      const double rating = datapoint(2);
+      itemMean(item) += rating;
+      ratingNum(item) += 1;
+    });
+
+    // Calculate item mean and subtract item mean from ratings.
+    // Set item mean to 0 if the item has no rating.
+    for (size_t i = 0; i < itemNum; i++)
+    {
+      if (ratingNum(i) != 0)
+        itemMean(i) /= ratingNum(i);
+    }
+
+    data.each_col([&](arma::vec& datapoint) {
```

Comment:
Can you put the opening ``{`` on a new line: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement.